### PR TITLE
fix: fund recovery

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

generates non-standard bitcoin and bitcoin addresses on a `/recovery` route with a copy to clipboard button

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

n/a

## Risk

none - not user facing

### Engineering

* visit `/recovery` with a native wallet, hit copy, paste into IDE, check that generated xpubs and addresses are different from standard addresses

### Operations

* not testable by operations

### CODE REVIEW

File: src/pages/Recovery/Recovery.tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+      <Button
+        color={copied ? 'green.500' : 'primary.500'}
+        isLoading={isLoading}
+        spinner={<Spinner size='md' />}
+        onClick={handleClick}
+        rightIcon={copied ? <FaCheck /> : undefined}
+      >
+        {copied ? 'Copied' : 'Copy to clipboard'}
+      </Button>
+      {isLocalhost && (
```
Comment (0xdef1cafe) [inline src/pages/Recovery/Recovery.tsx:214]: only spew this out to devs

File: src/pages/Recovery/Recovery.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+  const isLoading = !Boolean(dangerousAddresses.length)
+  const isLocalhost = window.location.hostname === 'localhost'
+
+  return (
+    <>
+      <Heading>
+        <Text translation='Recovery' />
+      </Heading>
+      <Button
```
Comment (0xdef1cafe) [inline src/pages/Recovery/Recovery.tsx:205]: a button they can use to copy/paste this payload back to support/ops

File: src/pages/Recovery/Recovery.tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+  for (const affectedChainId of affectedChainsIds) {
+    const maybeChainAdapter = chainAdapterManager.get(affectedChainId)
+    if (!maybeChainAdapter) throw new Error(`Recovery: no chain adapter for ${affectedChainId}`)
+    const adapter = maybeChainAdapter as unknown as UtxoBaseAdapter<UtxoChainId>
+    const chainName = adapter.getName()
+    for (const [bip44ParamChainName, wrongChainId] of Object.entries(KnownChainIds)) {
+      const bip44ParamsArray = dangerousBip44ParamsByChainId[wrongChainId]
+      for (const bip44Params of bip44ParamsArray) {
+        // only segwit native produces bc1-prefixed addresses
+        const accountType = UtxoAccountType.SegwitNative
```
Comment (0xdef1cafe) [inline src/pages/Recovery/Recovery.tsx:139]: i previously iterated all of the `UtxoAccountType`'s, but it generates `1` or `3` prefixed (legacy or segwit respectively) addresses, which are not what we want

File: src/pages/Recovery/Recovery.tsx [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+        const publicKey = await getDangerousPublicKey(wallet, bip44Params, accountType)
+        const addresses: string[] = []
+        // generate addresses
+        for (let index = 0; index < addressCount; index++) {
+          // only segwit native produces bc1-prefixed addresses
+          const scriptType = BTCInputScriptType.SpendWitness // segwit-native
+          const address = await wallet.btcGetAddress({
```
Comment (0xdef1cafe) [inline src/pages/Recovery/Recovery.tsx:148]: note for @kaladinlight so i don't forget, i think this is wrong/needs to be done at a lower level

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**